d2/src/main/java/com/linkedin/d2/jmx/RelativeLoadBalancerStrategyJmx.java

```diff
@@ -98,12 +119,17 @@ public double getMaxLatencyRelativeFactor()
   @Override
   public double getNthPercentileLatencyRelativeFactor(double pct)
   {
+    if (isPartitionDataUnavailable())
```

Comment:
Just to confirm, in a real situation, this function should always return true right?

Reply:
Correct, it will always return false.

Reply:
I just realized it can be a NPE if we always pass partition 0, but when it's using the first valid partition, there shouldn't be NPE anymore, but just to be careful I still added this check.